src/common/path/index.js

```diff
@@ -3,14 +3,17 @@ import { getIndex, isRange, isBranch } from '../node';
 export const PATH_SEPARATOR = '/';
 
 export function makePath(path) {
-  if (Array.isArray(path)) return path;
-  if (typeof path !== 'string') throw Error('resolve.path');
+  if (Array.isArray(path) && path.every(key => typeof key === 'string')) {
+    return path;
+  }
+  if (typeof path !== 'string') throw Error('makePath.path_not_string');
   if (!path.length || path === PATH_SEPARATOR) return [];
-  if (path[0] !== PATH_SEPARATOR) throw Error('resolve.path');
-  return path.split(PATH_SEPARATOR).slice(1);
+  path = path.split(PATH_SEPARATOR);
```

Comment:
Instead of reassigning the function argument, can we assign it to a constant and perform operations?